toolkit/tools/pkg/osmodifierlib/modifierutils.go

```diff
@@ -21,5 +26,163 @@ func doModifications(baseConfigPath string, osConfig *imagecustomizerapi.OS) err
 		return err
 	}
 
+	if osConfig.Overlays != nil {
+		_, err = imagecustomizerlib.EnableOverlays(osConfig.Overlays, dummyChroot)
+		if err != nil {
+			return err
+		}
+	}
+
+	if osConfig.KernelCommandLine.ExtraCommandLine != "" || osConfig.SELinux.Mode != "" {
+		bootCustomizer, err := imagecustomizerlib.NewBootCustomizer(dummyChroot)
+		if err != nil {
+			return err
+		}
+
+		err = handleSELinux(osConfig.SELinux.Mode, bootCustomizer, dummyChroot)
+		if err != nil {
+			return err
+		}
+
+		err = updateRootDevice(osConfig.KernelCommandLine.ExtraCommandLine, bootCustomizer)
+		if err != nil {
+			return err
+		}
+
+		err = bootCustomizer.WriteToFile(dummyChroot)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func modifyDefaultGrub() error {
+	var dummyChroot safechroot.ChrootInterface = &safechroot.DummyChroot{}
+	// Get verity and overlayfs values in /boot/grub2/grub.cfg
+	verityAndOverlayValues, rootDeviceValue, err := extractValuesFromGrubConfig(dummyChroot)
+	if err != nil {
+		return fmt.Errorf("error getting changes:\n%v", err)
+	}
+
+	bootCustomizer, err := imagecustomizerlib.NewBootCustomizer(dummyChroot)
+	if err != nil {
+		return err
+	}
+
+	// Stamp verity and overlayfs to /etc/default/grub
+	err = bootCustomizer.ApplyChangesToGrub("GRUB_CMDLINE_LINUX", verityAndOverlayValues)
```

Comment:
ah you're right, it could be problematic to replace entire GRUB_CMDLINE_LINUX. I should only update the certain values like selinux/verity/overlay ... and leave the rest unchanged

Reply:
add UpdateCmdlineValues  function which will:
1. Adding New Keys: If newValues contains a key that isn't already in GRUB_CMDLINE_LINUX, the function will add that key.
2. Updating Existing Keys: If the key already exists, the function will update its value.
3. Handling Flags: If a key is present without an = (e.g., nosplash), it's treated as a boolean flag and added accordingly.